src/Compatibility/ApiDiff/Microsoft.DotNet.ApiDiff/MemoryOutputDiffGenerator.cs

```diff
@@ -0,0 +1,669 @@
+        {
+            rightSideForLoopTasks.Add(Task.Run(async () =>
+            {
+                StringBuilder sb = new();
+                _log.LogMessage($"Visiting *new* assembly {afterAssemblySymbol.Name}...");
+
+                var swNew = Stopwatch.StartNew();
+                (SyntaxNode afterAssemblyNode, SemanticModel afterModel) = await GetAssemblyRootNodeAndModelAsync(_afterLoader, afterAssemblySymbol).ConfigureAwait(false);
+                string finalMessage = $"Finished visiting *new* {afterAssemblySymbol.Name} in {swNew.Elapsed.TotalMilliseconds / 1000.0:F2}s{Environment.NewLine}";
+
+                // We don't care about changed assembly attributes (for now).
+                swNew = Stopwatch.StartNew();
+                sb.Append(VisitChildren(beforeParentNode: null, afterAssemblyNode, beforeModel: null, afterModel, ChangeType.Inserted, wereParentAttributesChanged: false));
+                finalMessage += $"Finished generating diff for *new* assembly {afterAssemblySymbol.Name} in {swNew.Elapsed.TotalMilliseconds / 1000.0:F2}s{Environment.NewLine}";
+
+                _log.LogMessage(finalMessage);
+                swNew.Stop();
+
+                if (sb.Length > 0)
+                {
+                    _results.TryAdd(afterAssemblyName, GetFinalAssemblyDiff(afterAssemblyName, sb.ToString()));
+                }
+            }));
+        }
+
+        await Task.WhenAll(rightSideForLoopTasks).ConfigureAwait(false);
+
+        _log.LogMessage($"FINAL TOTAL: {swRun.Elapsed.TotalMilliseconds / 1000.0 / 60.0:F2} mins.");
+        swRun.Stop();
+    }
+
+    private async Task<(SyntaxNode rootNode, SemanticModel model)> GetAssemblyRootNodeAndModelAsync(IAssemblySymbolLoader loader, IAssemblySymbol assemblySymbol)
+    {
+        CSharpAssemblyDocumentGeneratorOptions options = new(loader, _symbolFilter, _attributeSymbolFilter)
+        {
+            HideImplicitDefaultConstructors = _hideImplicitDefaultConstructors,
+            ShouldFormat = false,
+            ShouldReduce = false,
+            MetadataReferences = loader.MetadataReferences,
+            DiagnosticOptions = _diagnosticOptions,
+            SyntaxRewriters = [
+                new TypeDeclarationCSharpSyntaxRewriter(_addPartialModifier), // This must be visited BEFORE GlobalPrefixRemover as it depends on the 'global::' prefix to be found
+                GlobalPrefixRemover.Singleton, // And then call this ASAP afterwards so there are fewer identifiers to visit
+                PrimitiveSimplificationRewriter.Singleton,
+                RemoveBodyCSharpSyntaxRewriter.Singleton,
+                AttributeNameSuffixRemover.Singleton,
+                SingleLineStatementCSharpSyntaxRewriter.Singleton,
+            ],
+            AdditionalAnnotations = [Formatter.Annotation] // Formatter is needed to fix some spacing
+        };
+
+        CSharpAssemblyDocumentGenerator docGenerator = new(_log, options);
```

Comment:
Here's how I create my `CSharpAssemblyDocumentGenerator` instance. Note the configuration options I chose.